packages/experimental/Avatar/src/JSAvatarTokens.ts

```diff
@@ -132,161 +132,132 @@ export const defaultJSAvatarTokens: TokenSettings<JSAvatarTokens, Theme> = (t: T
-      backgroundColor: globalTokens.color.purple.tint40,
-      color: globalTokens.color.purple.shade30,
-      borderColor: globalTokens.color.purple.primary,
+      ...getColorProps('purple', t),
     },
     grape: {
-      backgroundColor: globalTokens.color.grape.tint40,
-      color: globalTokens.color.grape.shade30,
-      borderColor: globalTokens.color.grape.primary,
+      ...getColorProps('grape', t),
     },
     lilac: {
-      backgroundColor: globalTokens.color.lilac.tint40,
-      color: globalTokens.color.lilac.shade30,
-      borderColor: globalTokens.color.lilac.primary,
+      ...getColorProps('lilac', t),
     },
     pink: {
-      backgroundColor: globalTokens.color.pink.tint40,
-      color: globalTokens.color.pink.shade30,
-      borderColor: globalTokens.color.pink.primary,
+      ...getColorProps('pink', t),
     },
     magenta: {
-      backgroundColor: globalTokens.color.magenta.tint40,
-      color: globalTokens.color.magenta.shade30,
-      borderColor: globalTokens.color.magenta.primary,
+      ...getColorProps('magenta', t),
     },
     plum: {
-      backgroundColor: globalTokens.color.plum.tint40,
-      color: globalTokens.color.plum.shade30,
-      borderColor: globalTokens.color.plum.primary,
+      ...getColorProps('plum', t),
     },
     beige: {
-      backgroundColor: globalTokens.color.beige.tint40,
-      color: globalTokens.color.beige.shade30,
-      borderColor: globalTokens.color.beige.primary,
+      ...getColorProps('beige', t),
     },
     mink: {
-      backgroundColor: globalTokens.color.mink.tint40,
-      color: globalTokens.color.mink.shade30,
-      borderColor: globalTokens.color.mink.primary,
+      ...getColorProps('mink', t),
     },
     platinum: {
-      backgroundColor: globalTokens.color.platinum.tint40,
-      color: globalTokens.color.platinum.shade30,
-      borderColor: globalTokens.color.platinum.primary,
+      ...getColorProps('platinum', t),
     },
     anchor: {
-      backgroundColor: globalTokens.color.anchor.tint40,
-      color: globalTokens.color.anchor.shade30,
-      borderColor: globalTokens.color.anchor.primary,
+      ...getColorProps('anchor', t),
     },
   } as JSAvatarTokens);
+
+/**
+ * A function which returns object of props depending on color and theme.
+ * @param color
+ * @param theme
+ * @returns object of props - backgroundColor, color and borderColor
+ */
+function getColorProps(color: string, theme: Theme) {
+  const themeAppearance = theme.host.appearance;
+  switch (themeAppearance) {
+    case 'light':
+    default:
+      return {
+        backgroundColor: globalTokens.color[color].tint40,
+        color: globalTokens.color[color].shade30,
+        borderColor: globalTokens.color[color].primary,
+      };
+    case 'dark':
+      return {
+        backgroundColor: globalTokens.color[color].shade30,
+        color: globalTokens.color[color].tint40,
+        borderColor: globalTokens.color[color].tint30,
+      };
+    case 'highContrast':
```

Comment:
In the spec/Figma it looks like there's both a border and a ring, and they are separate tokens. I think with the current implementation if the Avatar has no ring, then in HC there is no visible border around the Avatar, which would make it difficult for users to see where the Avatar is (they would just see floating initials). It looks like we might need to have both a border and also a ring.

Reply:
Yes, it's a known bug. I returned ringColor token. I haven't finished it yesterday. I'll try to fix it today